rust/log/src/types.rs

```diff
@@ -15,68 +15,18 @@ pub struct CollectionInfo {
     pub first_log_ts: i64,
 }
 
-#[derive(Error, Debug)]
-pub enum PullLogsError {
-    #[error("Failed to fetch")]
-    FailedToPullLogs(#[from] tonic::Status),
-    #[error("Failed to convert proto embedding record into EmbeddingRecord")]
-    ConversionError(#[from] RecordConversionError),
-}
-
-impl ChromaError for PullLogsError {
-    fn code(&self) -> ErrorCodes {
-        match self {
-            PullLogsError::FailedToPullLogs(e) => e.code().into(),
-            PullLogsError::ConversionError(_) => ErrorCodes::Internal,
-        }
-    }
-}
-
-#[derive(Error, Debug)]
-pub enum PushLogsError {
-    #[error("Failed to push logs")]
-    FailedToPushLogs(#[from] tonic::Status),
-    #[error("Failed to convert records to proto")]
-    ConversionError(#[from] RecordConversionError),
-}
-
-impl ChromaError for PushLogsError {
-    fn code(&self) -> ErrorCodes {
-        match self {
-            PushLogsError::FailedToPushLogs(_) => ErrorCodes::Internal,
-            PushLogsError::ConversionError(_) => ErrorCodes::Internal,
-        }
-    }
-}
-
-#[derive(Error, Debug)]
-pub enum GetCollectionsWithNewDataError {
-    #[error("Failed to fetch")]
-    FailedGetCollectionsWithNewData(#[from] tonic::Status),
-}
-
-impl ChromaError for GetCollectionsWithNewDataError {
-    fn code(&self) -> ErrorCodes {
-        match self {
-            GetCollectionsWithNewDataError::FailedGetCollectionsWithNewData(_) => {
-                ErrorCodes::Internal
-            }
-        }
-    }
-}
-
-#[derive(Error, Debug)]
-pub enum UpdateCollectionLogOffsetError {
-    #[error("Failed to update collection log offset")]
-    FailedToUpdateCollectionLogOffset(#[from] tonic::Status),
-}
+/// Implements `ChromaError` for `sqlx::Error`.
+#[derive(Debug, Error)]
+#[error("Database error: {0}")]
+pub struct WrappedSqlxError(pub sqlx::Error);
```

Comment:
I think if we want to introduce this we should maybe put it in sqlite/ and use it from there in all uses of our sqlite + sqlx